### DISCUSSION TITLE
How to visualize instance predictions without class label?

### DISCUSSION THREAD

**Discussion by ck12780** [facebookresearch/detectron2#5262] [Q&A]
![image](https://github.com/facebookresearch/detectron2/assets/155506569/bace27fd-2f35-40e0-841a-4aa0c4a2f65f)
![image](https://github.com/facebookresearch/detectron2/assets/155506569/c6598e02-7c42-47b7-90ee-bbb1995c05bc)
Image 1 is the instance predictions done after training detectron2 model, image 2 is the binary image of image 1. In order to calculate number of pixel in the segmented area, i need to remove the class labels and the predictions percentage in the picture. How to do this? Annotations done using ROBOFLOW to download coco.json file

`import os
import cv2
import matplotlib.pyplot as plt
from detectron2.data import DatasetCatalog
from detectron2.utils.visualizer import Visualizer, ColorMode

predicted_images_path = os.path.abspath("/content/predicted")
dataset_dicts_validation = DatasetCatalog.get('void-detection-3-valid')

for d in dataset_dicts_validation:
    im = cv2.imread(d["file_name"])
    outputs = predictor(I'm)

    v = Visualizer(im[:, :, ::-1], metadata=metadata, scale=0.5, instance_mode=ColorMode.SEGMENTATION)
    out = v.draw_instance_predictions(outputs["instances"].to("cpu"))

    cv2.imwrite(f"{predicted_images_path}/{d['file_name'].split('/')[-1]}", out.get_image()[:, :, ::-1])`

**Comment by AliAbdullahTech** [comment]
Go to "visualizer.py", remove the labels from the overlay_instances function call from draw_instance_predictions function. 

Edit:
Better yet add parameter in the draw instance function which has a boolean value and make changes in function which will make value of label None if boolean is false.

**Comment by teffuate-coder** [comment]
![Uploading IMG_0954.png…]()
